FeedStoreChallenge/CoreDataStore/CoreDataStore.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Created by Shady
+//  All rights reserved.
+//  
+
+import CoreData
+
+open class CoreDataStore : FeedStore{
+    enum ModelError : Error {
+        case modelNotFound
+    }
+    private let container: NSPersistentContainer
+    private let context: NSManagedObjectContext
+
+    private static let modelName = "FeedModel"
+    private static let model = NSManagedObjectModel.with(name: modelName, in: Bundle(for: CoreDataStore.self))
+
+    public init(storeURL:URL,storeName:String) throws{
+        guard let model = CoreDataStore.model else {
+            throw ModelError.modelNotFound
+        }
+        do {
+            container = try NSPersistentContainer.load(name: storeName, model: model, url:storeURL)
+            context = container.newBackgroundContext()
+        }
+        catch{
+            throw error
+        }
```

Comment:
Since the `init` method throws, you don't need a do/catch block here. The error will propagate directly.

```suggestion
        container = try NSPersistentContainer.load(name: storeName, model: model, url:storeURL)
        context = container.newBackgroundContext()
```